src/runtime/virtcontainers/clh.go

```diff
@@ -908,7 +930,9 @@ func (clh *cloudHypervisor) Capabilities(ctx context.Context) types.Capabilities
 
 	clh.Logger().WithField("function", "Capabilities").Info("get Capabilities")
 	var caps types.Capabilities
-	caps.SetFsSharingSupport()
+	if !clh.config.ConfidentialGuest {
+		caps.SetFsSharingSupport()
+	}
```

Comment:
Call me pedantic, but I would add this blob and the one where you define `supportsSharedFS()` in a separate commit. Just to make things easier to follow.

Reply:
Addressed!